idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/utils/IdeUtils.kt

```diff
@@ -185,6 +189,41 @@ fun PsiElement.ctx(): CompilerContext? =
 fun Project.ctx(): CompilerContext? =
   getService(CompilerContext::class.java)
 
+fun <P : PsiElement> AnnotationBuilder.registerLocalFix(
+  fix: LocalQuickFix,
+  psi: P,
+  message: String,
+  highlightType: ProblemHighlightType = ProblemHighlightType.INFORMATION,
+  onThefly: Boolean = true,
+  afterBorderLine: Boolean = true
+): AnnotationBuilder.FixBuilder =
+  newLocalQuickFix(fix, psi.inspectionManager()
+    .createProblemDescriptor(psi, message, arrayOf(fix), highlightType, onThefly, afterBorderLine))
+
+fun <P : PsiElement> descriptor(
+  element: P,
+  message: String,
+  fix: LocalQuickFix? = null,
+  highlightType: ProblemHighlightType = ProblemHighlightType.INFORMATION,
+  onThefly: Boolean = true
+): ProblemDescriptor =
+  element.inspectionManager().createProblemDescriptor(element, message, fix, highlightType, onThefly)
+
+fun PsiElement.inspectionManager(): InspectionManager =
+  InspectionManager.getInstance(project)
+
+fun localQuickFix(name: String, familyName: String, f: Project.(ProblemDescriptor) -> Unit): LocalQuickFix =
```

Comment:
```suggestion
fun localQuickFix(name: String, message: String, f: Project.(ProblemDescriptor) -> Unit): LocalQuickFix =
```
this is maybe a more intuitive signature. 
Given we need `name`. But It's up to you :D

Reply:
yeah much better 👍

Reply:
`name` can then be removed, right?

Reply:
yes!